backend/app/routers/experiments.py

```diff
@@ -47,27 +45,32 @@ async def get_experiments(
     response_model=int,
 )
 async def get_experiments_count(
-    user: Json = Depends(get_current_user_optional),
-    include_mine: bool = False,
-    query_operator: QueryOperator = QueryOperator.AND,
+    user: Annotated[dict, Depends(get_current_user(required=False))],
 ) -> Any:
-    result_set = find_specific_experiments(
-        include_mine=include_mine, query_operator=query_operator, user=user
-    )
-    return await result_set.count()
+    # TODO hotfix for returning only my experiments
+    if user:
+        return await Experiment.find(Experiment.created_by == user["email"]).count()
+    else:
+        return 0
 
 
 @router.get(
     "/experiments/{id}",
     response_model=ExperimentResponse,
 )
-async def get_experiment(id: PydanticObjectId) -> Any:
+async def get_experiment(
+    id: PydanticObjectId,
+    user: Annotated[dict, Depends(get_current_user(required=False))],
+) -> Any:
     experiment = await Experiment.get(id)
+
     if experiment is None:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail="Specified experiment doesn't exist",
         )
+
+    await check_experiment_access(id, user)
```

Comment:
It would be beneficial to have it more explicit. There are multiple options:
1.
assert await check_experiment_access(id, user) == true, ...

2.
await raise_exception_if_no_experiment_access(id, user) # ugly name, but you get the point)

Reply:
I am also not a big fan of calling function in this way, but I could not find the best way to to it here. I assume this will be changed once we have a proper access management in place. But from your options I would vote for 2 as it is the most explicit I guess.. I will think about it a bit